lms/djangoapps/instructor_task/api.py

```diff
@@ -437,14 +437,18 @@ def submit_export_ora2_data(request, course_key):
     return submit_task(request, task_type, task_class, course_key, task_input, task_key)
 
 
-def generate_certificates_for_students(request, course_key, students=None):  # pylint: disable=invalid-name
+def generate_certificates_for_students(request, course_key, student_set, specific_student):  # pylint: disable=invalid-name
     """
     Submits a task to generate certificates for given students enrolled in the course or
     all students if argument 'students' is None
 
     Raises AlreadyRunningError if certificates are currently being generated.
     """
-    if students:
+    if student_set:
+        task_type = 'generate_certificates_student_set'
+        task_input = {'students': students}
```

Comment:
This is going to get confusing really fast unless we rename the dictionary keys depending upon if we're passing a specific student or a student set. If we're passing a `student_set`, we should use `student_set` as the key. If we're passing a specific student id, we should be passing `specific_student_id` as a string.